### PR TITLE
Hosting Command Palette: Add loading notices to actions

### PR DESCRIPTION
**Pull request by sejas**
- Related to https://github.com/Automattic/dotcom-forge/issues/4707

## Proposed Changes

* Display a loading notice when taking an action.

## Testing Instructions
1. Go to `/sites` or any other page that our command palette is available
2. Press `cmd+k` on Mac and `ctrl+k` on Windows
3. Search for `Reset SSH/SFTP password` and press enter
4. Observe a temporal loading notice 
5. Observe a second success notice after a few seconds
6. Repeat step 3 searching for `Copy SSH connection string`

## Screencast

https://github.com/Automattic/wp-calypso/assets/779993/a1407a40-324a-4da5-a5c5-5e986c2eac68

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by katinthehatsite — APPROVED** [review]
It works as expected! Nice work!

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 1]
```diff
@@ -101,11 +120,13 @@ export const useCommandsArrayWpcom = ( {
...
 
 		if ( ! sshPassword ) {
+			removeLoadingNotice();
 			return navigate( `/hosting-config/${ siteSlug }` );
 		}
 
 		navigator.clipboard.writeText( sshPassword );
-		displaySuccessNotice( __( 'Copied new password' ) );
+		removeLoadingNotice();
+		displayNotice( __( 'The new password was copied to your clipboard' ) );
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx]:
```suggestion
		displayNotice( __( 'SSH/SFTP password reset and copied to clipboard.' ) );
```

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 2]
```diff
@@ -73,20 +82,30 @@ export const useCommandsArrayWpcom = ( {
...
 		if ( ! sshUser ) {
+			removeLoadingNotice();
 			return navigate( `/hosting-config/${ siteSlug }` );
 		}
 
 		const textToCopy = copyType === 'username' ? sshUser : `ssh ${ sshUser }@sftp.wp.com`;
 		navigator.clipboard.writeText( textToCopy );
+		removeLoadingNotice();
 		const successMessage =
 			copyType === 'username' ? __( 'Copied username' ) : __( 'Copied SSH connection string' );
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx]:
```suggestion
			copyType === 'username' ? __( 'Copied username.' ) : __( 'Copied SSH connection string.' );
```

**Review by danielbachhuber — APPROVED** [review]
Much better! A couple suggested string changes, but looks good otherwise.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9809107

Some locales (Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @sejas, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Copying username…`
- `Resetting SSH/SFTP password…`
- `SSH/SFTP password reset and copied to clipboard.`
- `Copying SSH connection string…`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9809107).

**Pull request merged**